packages/grid/_modules_/grid/hooks/features/rows/useGridEditRows.ts

```diff
@@ -45,6 +47,21 @@ export function useGridEditRows(apiRef: GridApiRef) {
   const [, setGridState, forceUpdate] = useGridState(apiRef);
   const options = useGridSelector(apiRef, optionsSelector);
   const lastEditedCell = React.useRef<GridCellParams | null>(null);
+  const insideCell = React.useRef(false);
+
+  const handleDocumentClick = useEventCallback(() => {
+    const params = lastEditedCell.current;
+    if (insideCell.current || !params) {
+      insideCell.current = false;
+      return;
+    }
+    const cellCommitParams = apiRef.current.getEditCellPropsParams(params.id, params.field);
+    if (!cellCommitParams.props.error) {
+      // We commit the change when there are no error
+      apiRef.current.publishEvent(GRID_CELL_EDIT_PROPS_CHANGE_COMMITTED, cellCommitParams);
```

Comment:
I can't pass an event because `publishEvent` requires a `SyntheticEvent` (because of the `isPropagationStopped`) and I only have a native event.

Reply:
Could we make this generic? Otherwise, there would be portal fixes in a lot of features, and each hook will attach and detach events...
If we add an event listener on the doc, and route it to publish some kind of `CLICK_OUTSIDE_ROOT`
And isolate that into a usePortal hook, it would be really nice.
Then each feature, can hook themselves to some kind of click on portal event...
What do you think?

Reply:
The click we're interested may occur inside the root as well. What we need to check is if it happened outside the cell that is currently being edited. Since it's a very specific situation, I don't think we need to abstract it. This new event would have to be named like `GRID_EDIT_CELL_CLICK_OUTSIDE` and it would be attached only to the edit cell, which is the same thing I doing here but without exposing an event.

Reply:
what if we add a renderCell that use portals? Ie a select component in the grid view mode

Reply:
> what if we add a renderCell that use portals? Ie a select component in the grid view mode

We wouldn't have problems because clicking outside the cell in view mode doesn't do anything besides moving the focus.

Reply:
unless there is a select dropdown within it

Reply:
Here's a CodeSandbox with a Select in `renderCell`: https://codesandbox.io/s/material-demo-forked-kc7id?file=/demo.js

We only have a bug if a portaled component is used in `renderEditCell`.